nannyml/drift/ranking.py

```diff
@@ -200,3 +260,118 @@ def rank(
         if only_drifting:
             ranking = ranking.loc[ranking['number_of_alerts'] != 0, :]
         return ranking
+
+@Ranker.register('correlation')
+class CorrelationRanking(Ranking):
+    """Ranks features according to the correlation of their drift results and
+    absolute performance change from mean reference performance.
+    """
+
+
+    def fit(
+        self,
+        performance_results: Union[
+            CBPEResults, DLEResults, PerformanceCalculationResults
+        ],
+    ):
+        """
+        Use performance results from the reference period in order to be able to compare drift
+        results with absolute performance changes from the mean performance results during the reference period.
+
+        Parameters
+        ----------
+        performance_results : Performance Estimation or Performance Calculation results filtered to
+            the performance metric we want to compare against drift results. Additionally results need
+            to be filtered to the reference period only.
+            Can be an instance of:
+            nml.performance_estimation.confidence_based.results.Result,
+            nml.performance_estimation.direct_loss_estimation.result.Result,
+            nml.performance_calculation.result.Result
+        """
+
+        # User should filter
+        metrics = performance_results.metrics
+        if len(metrics) > 1:
+            raise ValueError("Only one metric should be present in performance_results used to fit CorrelationRanking.")
```

Comment:
Maybe use the `nannyml.exceptions.InvalidArugmentsException`

Reply:
Addressed with 55e6f9be4140ee7025dcccd3b423c9508bdb236f